### PR TITLE
run_ps output can't be logged to xcat.log file

### PR DESCRIPTION
**Pull request by penguhyang**
fix issue: #547

### CODE REVIEW

**Comment by whowutwut** [comment]
@penguhyang I wonder if xCAT should be using rsyslog to do logging into the files....

Basically creating a policy in `/etc/rsyslog.d/` and using the logger function...  instead of using `tee -a`?

**Comment by immarvin** [comment]
hi @whowutwut , run_ps is called in post.xcat and other post-installation scripts in the installer before "syslog" postscript is run, at this time. the rsyslog might not be running and configured well, we cannot rely on it

**Comment by immarvin** [comment]
hi @penguhyang , still cannot get what's the root cause of the defect and why your fix will fix it, please explain more

File: xCAT-server/share/xcat/install/scripts/post.xcat [hunk 1]
```diff
@@ -178,7 +178,7 @@ run_ps () {
      local compt=\$(file \$1)
      local reg=\"shell script\"
      if [[ \"\$compt\" =~ \$reg ]]; then
-        bash -x ./\$@ 2>&1
+        bash -x ./\$@ 2>&1 | tee -a \$logfile |  logger -t xcat -p debug
```
Comment (immarvin) [inline xCAT-server/share/xcat/install/scripts/post.xcat]: "bash -x ./\$@ 2>&1" should be kept if you have fixed the msgutils_r in xcatlib.sh

File: xCAT/postscripts/xcatlib.sh [hunk 2]
```diff
@@ -718,7 +718,7 @@ function msgutil_r {
    if [ -n "$logserver" ];then
       logger -n $logserver -t xcat -p local4.$msgtype "$msgstr" >/dev/null 2>&1
       if [ "$?" != "0" ];then
-         exec 3<>/dev/udp/$logserver/514 >/dev/null 2>&1;logger -s -t xcat -p local4.$msgtype "$msgstr" 1>&3  2>&1
+         exec 3<>/dev/udp/$logserver/514 >/dev/null;logger -s -t xcat -p local4.$msgtype "$msgstr" 1>&3  2>&1
```
Comment (immarvin) [inline xCAT/postscripts/xcatlib.sh]:
````
exec 3<> /dev/udp/$logserver/514 && logger -s -t xcat -p local4.$msgtype "$msgstr" 1>&3  2>&1
````

Comment (penguhyang) [inline xCAT/postscripts/xcatlib.sh]: @immarvin Thanks for review. Code updated.

File: xCAT/postscripts/xcatlib.sh [hunk 3]
```diff
@@ -718,7 +718,7 @@ function msgutil_r {
    if [ -n "$logserver" ];then
       logger -n $logserver -t xcat -p local4.$msgtype "$msgstr" >/dev/null 2>&1
       if [ "$?" != "0" ];then
-         exec 3<>/dev/udp/$logserver/514 >/dev/null 2>&1;logger -s -t xcat -p local4.$msgtype "$msgstr" 1>&3  2>&1
+         exec 3<>/dev/udp/$logserver/514 && logger -s -t xcat -p local4.$msgtype "$msgstr" 1>&3  2>&1
```
Comment (immarvin) [inline xCAT/postscripts/xcatlib.sh]: the file descriptor 3 should be closed, 
````
exec 3<>/dev/udp/$logserver/514 && logger -s -t xcat -p local4.$msgtype "$msgstr" 1>&3  2>&1 && exec 3>&-
````

Comment (penguhyang) [inline xCAT/postscripts/xcatlib.sh]: fixed.

**Pull request merged**